clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -705,6 +743,25 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::init(
 {
     options_ = std::move(options);
     verbosity_ = options_.verbosity;
+
+    // Add noise generator reader to workload_inputs.
+    if (options_.noise_generator_add > 0) {
+        for (uint64_t noise_generator_idx = 0;
+             noise_generator_idx < options_.noise_generator_add; ++noise_generator_idx) {
+            auto noise_generator =
+                get_noise_generator(static_cast<addr_t>(noise_generator_idx + 1),
+                                    static_cast<addr_t>(noise_generator_idx + 1),
+                                    options_.noise_generator_num_records);
+            auto noise_generator_end = get_noise_generator_end();
+            std::vector<typename sched_type_t::input_reader_t> readers;
+            // Use a sentinel for the tid so the scheduler will use the memref record tid.
```

Comment:
But you know the tid, don't you?

Reply:
Yes, now setting the same tid of the TRACE_TYPE_THREAD record produced by the noise generator.